units/provisioning_create_host.go

```diff
@@ -144,14 +145,50 @@ func (j *createHostJob) Run(ctx context.Context) {
 			grip.Error(message.WrapError(err, message.Fields{
 				"host_id":  j.HostID,
 				"attempt":  j.CurrentAttempt,
-				"distro":   j.host.Distro,
+				"distro":   j.host.Distro.Id,
 				"job":      j.ID(),
 				"provider": j.host.Provider,
 				"message":  "could not remove intent document",
 				"outcome":  "host pool may exceed maxhost limit",
 			}))
 
 			return
+		} else if !j.host.IsSubjectToHostCreationThrottle() {
+			// pass (spawn hosts and friends should always
+			// try and start)
+		} else if j.host.Status == evergreen.HostBuilding {
+			// pass (containers or similar that are
+			// building should continue to provision)
+		} else {
+			var (
+				numProv            int
+				runningHosts       int
+				distroRunningHosts int
+			)
+
+			numProv, err = host.CountProvisioning()
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+			distroRunningHosts, err = host.CountRunningHosts(j.host.Distro.Id)
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+
+			runningHosts, err = host.CountRunningHosts("")
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+
+			if distroRunningHosts < runningHosts/100 || distroRunningHosts < j.host.Distro.HostAllocatorSettings.MinimumHosts {
+				// pass
+			} else if numProv >= j.env.Settings().HostInit.HostThrottle {
```

Comment:
Task-spawned hosts and spawn hosts can starve task hosts, which is why there is this check in the cron job https://github.com/evergreen-ci/evergreen/pull/3492/files#diff-5d21b8dfd83d471b34c22f4156c931aaR886-R888. Should `provisioningHostsQuery` look only for task hosts? I think it could look at things spawned by the Evergreen user only.

Reply:
hasn't this always been the case, also they could (theoretically) count against maxhosts?

Reply:
No. The crons throttle only increments for task hosts https://github.com/evergreen-ci/evergreen/blob/f1f7e10ed5408cbd4a5b57db802437b216558d06/units/crons.go#L889.